providers/gavc_download.rb

```diff
@@ -14,6 +17,9 @@ def whyrun_supported?
                                                  packaging: new_resource.packaging)
 
   resource_name = "Downloading file #{download_uri} specified by: #{new_resource.to_s}"
+  if new_resource.username && new_resource.password then
+    download_uri.sub!(/^(.*):\/\//, "\\1://#{new_resource.username}:#{new_resource.password}@")
```

Comment:
ah sorry. I don't know, how I write a unit test for this. my chef experience isn't so good at this time...